Src/Particle/AMReX_Particle.H

```diff
@@ -15,13 +15,14 @@ namespace amrex {
 
 namespace
 {
-    /** Used for 64bit Long particle Ids as in AoS layout */
+    /** Special flags used for 64-bit Long particle Ids */
     namespace LongParticleIds {
```

Comment:
wouldn't it be better for these to be `uint64_t` instead of `amrex::Long`, to match `m_idcpu`?

Reply:
Good point, yes.

@AlexanderSinn, do you agree or do we overlook something?

Reply:
Since `ptd.idcpu()` is supposed to be used with `InvalidParticleID` (type uint64_t) and `ptd.id()` should be used with the other special ids (type amrex::Long), I think `InvalidParticleID` should be in a separate namespace with type `uint64_t` while the other special IDs remain how they are. Maybe it can be shortened to `ParticleIDCPUs::Invalid`.

Once all locations that make particles invalid are updated to this, it would allow for more efficient checks if a particle is invalid `if (ptd.idcpu(ip) == ParticleIDCPUs::Invalid)` and the use of a 40-bit id instead of a 39-bit one.

Reply:
Not sure about the last part, because what counts is memory alignment for transfer, which will be 64bit, but agree on everything else. Thanks!